### PR TITLE
vencord: 1.4.7 -> 1.5.3

### PR DESCRIPTION
**Pull request by fwam**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yayayayaka** [comment]
Result of `nixpkgs-review pr 257701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>vencord</li>
    <li>vencord-web-extension</li>
    <li>webcord-vencord</li>
  </ul>
</details>

**Comment by NotAShelf** [comment]
Result of `nixpkgs-review pr 257701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>vencord</li>
    <li>vencord-web-extension</li>
    <li>webcord-vencord</li>
  </ul>
</details>

**Review by NotAShelf — APPROVED** [review]

**Comment by NotAShelf** [comment]
Result of `nixpkgs-review pr 257701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>vencord</li>
    <li>vencord-web-extension</li>
    <li>webcord-vencord</li>
  </ul>
</details>

**Review by NotAShelf — APPROVED** [review]

**Review by NotAShelf — APPROVED** [review]
shut up ofborg

File: pkgs/misc/vencord/default.nix [hunk 1]
```diff
@@ -36,13 +36,15 @@ buildNpmPackage rec {
   # Supresses an error about esbuild's version.
   npmRebuildFlags = [ "|| true" ];
 
-  npmDepsHash = "sha256-GoVVOLg20oi0MJGLqevpiqHDM/7yaRJSQnM/tt+AkQ8=";
+  makeCacheWritable = true;
```
Comment (Scrumplex) [inline pkgs/misc/vencord/default.nix:39]: Why is this needed?

Comment (fwam) [reply pkgs/misc/vencord/default.nix:39]: otherwise it has the issues with writing to cache, basically.

Comment (fwam) [reply pkgs/misc/vencord/default.nix:39]:
```
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /nix/store/qlps29dh4gavnccma4k8sa61jhqq36py-vencord-1.5.3-npm-deps/_cacache/tmp
npm ERR! errno -13
npm ERR! 
npm ERR! Your cache folder contains root-owned files, due to a bug in
npm ERR! previous versions of npm which has since been addressed.
npm ERR! 
npm ERR! To permanently fix this problem, please run:
npm ERR!   sudo chown -R 1000:100 "/nix/store/qlps29dh4gavnccma4k8sa61jhqq36py-vencord-1.5.3-npm-deps"
```

Comment (Scrumplex) [reply pkgs/misc/vencord/default.nix:39]: I am wondering why this is needed now. How did you generate your updated package-lock.json?

Comment (fwam) [reply pkgs/misc/vencord/default.nix:39]: through `npm i` on `tags/v1.5.3` of `Vendicated/Vencord`

Comment (Scrumplex) [reply pkgs/misc/vencord/default.nix:39]: Could you try regenerating it using `npm install --legacy-peer-deps -f`? That's what @FlafyDev and I have been doing IIRC

Comment (fwam) [reply pkgs/misc/vencord/default.nix:39]: > Could you try regenerating it using `npm install --legacy-peer-deps -f`? That's what @FlafyDev and I have been doing IIRC

ah, my bad, `--legacy-peer-deps` was used, too. didn't use `-f` tho

**Review by thiagokokada — APPROVED** [review]

**Comment by FlafyDev** [comment]
I don't think this should've been merged yet.. 
ofborg couldn't build it and I also can't build it with `nix build github:NixOS/nixpkgs#vencord` or `nix build github:fwam/nixpkgs/vencord#vencord`

https://github.com/NixOS/nixpkgs/runs/17211974849

**Comment by thiagokokada** [comment]
> I don't think this should've been merged yet.. ofborg couldn't build it and I also can't build it with `nix build github:NixOS/nixpkgs#vencord` or `nix build github:fwam/nixpkgs/vencord#vencord`
> 
> https://github.com/NixOS/nixpkgs/runs/17211974849

I thought this was just OfBorg misbehaving, but if this is not the case can you open a revert PR? I can merge it.

**Comment by fwam** [comment]
Check #257805

**Comment by K900** [comment]
Added backport label on request from @fwam.
